server/to.js

```diff
@@ -1,9 +1,11 @@
-const { action }      = require('@articulate/ducks')
-const { curryN, tap } = require('ramda')
+const { curry } = require('ramda')
 
-// to : Socket -> (a -> String) -> String -> a -> a
-const to = curryN(3, (socket, room, type) => tap(payload =>
-  socket.broadcast.to(room(payload)).emit('action', action(type, payload))
-))
+const cleanMeta = require('./cleanMeta')
 
-module.exports = to
+// to :: (Action -> String) -> Action -> Action
+const to = (room, axn) => {
+  axn.meta.socket.broadcast.to(room(axn)).emit('action', cleanMeta(axn))
```

Comment:
Additionally... we lose the ability to broadcast to `io`. We _must_ broadcast to `socket`, which will always exclude the current user. This will break a lot of our patterns.

Reply:
The most common use case that I've seen is to broadcast to all sockets except the current one.  In cases where we broadcast without exception, I figured it'd be easy enough to do this:
```js
const io = require('socket.io-emitter')(process.env.REDIS_URI)
const { assocPath } = require('ramda')

const { adminRoom } = require('../lib/rooms')

const someAsyncWork = pipeP(
  // the real work, producing an action
  assocPath(['meta', 'socket'], io),
  to(adminRoom)
)
```